l2geth/internal/ethapi/api.go

```diff
@@ -934,17 +934,19 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 		executionManager := cfg.StateDump.Accounts["OVM_ExecutionManager"]
 		stateManager := cfg.StateDump.Accounts["OVM_StateManager"]
 		block, err := b.BlockByNumber(ctx, rpc.BlockNumber(header.Number.Uint64()))
-		if err != nil {
-			return nil, 0, false, err
-		}
-		txs := block.Transactions()
-		if header.Number.Uint64() != 0 {
-			if len(txs) != 1 {
-				return nil, 0, false, fmt.Errorf("block %d has more than 1 transaction", header.Number.Uint64())
+		if block != nil {
+			if err != nil {
+				return nil, 0, false, err
+			}
```

Comment:
This should still be checked every time, independently of the `block` not being nil